app.js

```diff
@@ -1,64 +1,105 @@
 const productsCenterDOM = document.querySelector(".products-center");
-const cartBtn = document.querySelector(".cart-btn");
 const cart = document.querySelector(".cart");
+const cartContainer = document.querySelector(".cart-content");
 const cartOverlay = document.querySelector(".cart-overlay");
-const closeCartIcon = document.querySelector(".close-cart");
-let allChevronUpBtns = document.querySelectorAll(".fa-chevron-up");
-let allChevronDownBtns = document.querySelectorAll(".fa-chevron-down");
 let allitemAmounts = document.querySelectorAll(".item-amount");
 
 (async function () {
   let response = await fetch("products.json");
-  let data = await response.json();
+  let allProducts = await response.json();
   let results = "";
 
-  for (let index = 0; index < data.length; index++) {
+  for (let index = 0; index < allProducts.length; index++) {
     results += `
        <article class="product">
           <div class="img-container">
               <img
-                  src="${data[index].url}"
+                  src="${allProducts[index].url}"
                   alt="product"
                   class="product-img"
               />
               <button class="bag-btn" data-id="1">
                   <i class="fas fa-shopping-cart"></i>
-                  add to bag
+                  add to cart
               </button>
           </div>
-          <h3>${data[index].name}</h3>
-          <h4>$${data[index].price}</h4>
+          <h3>${allProducts[index].name}</h3>
+          <h4>$${allProducts[index].price}</h4>
       </article>
   `;
   }
   productsCenterDOM.innerHTML = results;
+
+  const allAddToCartBtns = document.querySelectorAll(".bag-btn");
```

Comment:
Why have you written this code inside IIFE ? It could have worked fine outside this function.